index.js

```diff
@@ -175,12 +177,23 @@ app.use(wkj);
 app.use(oadaError());
 
 if (config.server.httpMode) {
-  var server = app.listen(process.env.PORT || config.server.port, function() {
+   var server = app.listen(process.env.PORT || config.server.port, function() {
     console.log('Listening HTTP on port %d', server.address().port);
   });
 } else {
   var server = https.createServer(config.certs, app);
+
+  var shutdown = function(){
+    console.log("Gracefully closing")
+    server.close();
+  }
+  //this is the signal that kill generates
+  process.on('SIGTERM', shutdown);
+  //this is the signal that ctrl-c generates
+  process.on('SIGINT',  shutdown);
+
```

Comment:
I am okay with this but please move it out of the if statement so that it applies for both http and https